tests/components/vesync/test_init.py

```diff
@@ -98,3 +106,213 @@ async def test_async_setup_entry__loads_fans(
     assert hass.data[DOMAIN][VS_FANS] == [fan]
     assert not hass.data[DOMAIN][VS_LIGHTS]
     assert hass.data[DOMAIN][VS_SENSORS] == [fan]
+
+
+async def test_async_process_devices__no_devices(
```

Comment:
Please add the tests in a separate PR that we can merge first. Then we can make sure that the refactor works without changing any tests.